website/docs/public/netlify.toml

```diff
@@ -0,0 +1,4 @@
+[[redirects]]
+from = "/*"
+to = "/index.html"
+status = 404
```

Comment:
404 status code is consistent with github
<img width="400" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/79413249/d521d5cd-fd6e-4de7-8274-7d4a2b717da0">

Reply:
The 404 status code is estimated that the search engine will not index it because it is an invalid page. When it returns 200, it will consider it a meaningful page and then index it. However, the page content of 404 is all the same, and it will think that there are relatively more duplicate pages on your website, and the weight may be reduced.